src/components/shared/ReconfigurationStrategyDrawer.tsx

```diff
@@ -0,0 +1,158 @@
+    initialSelectedStrategy,
+  )
+
+  const selectedStrategyIndex = ballotStrategies().findIndex(s => {
+    return s.address?.toLowerCase() === selectedStrategy.address?.toLowerCase()
+  })
+
+  const [customStrategyAddress, setCustomStrategyAddress] = useState<string>(
+    selectedStrategyIndex === -1 ? initialSelectedStrategy.address : '', // only set if selected strategy is custom
+  )
+
+  return (
+    <Drawer visible={visible} {...style} onClose={onClose}>
+      <Space direction="vertical">
+        <h1>
+          <Trans>Reconfiguration</Trans>
+        </h1>
+
+        <div>
+          <p style={{ color: colors.text.secondary }}>
+            <Trans>
+              Rules for how this project's funding cycles can be reconfigured.
+            </Trans>
+          </p>
+
+          {ballotStrategies()[selectedStrategyIndex]?.address ===
+            ballotStrategies()[0].address && (
+            <p style={{ color: colors.text.warn }}>
+              <Trans>
+                Using a reconfiguration strategy is recommended. Projects with
+                no strategy will appear risky to contributors.
+              </Trans>
+            </p>
+          )}
+        </div>
+        {ballotStrategies().map((s: Strategy, i) => (
+          <ReconfigurationStrategyOption
+            title={s.name}
+            key={i}
+            content={
+              <div>
+                <p>{s.description}</p>
+                <p style={{ fontSize: '0.7rem', color: colors.text.tertiary }}>
+                  <Trans>Contract address: {s.address}</Trans>
+                </p>
+              </div>
+            }
+            strategy={s}
+            selected={selectedStrategyIndex === i}
+            onSelectBallot={() => setSelectedStrategy(s)}
+            index={i}
+          />
+        ))}
+        <ReconfigurationStrategyOption
+          title={t`Custom strategy`}
+          content={
+            <CustomStrategyForm
+              address={customStrategyAddress}
+              setAddress={(address: string) => {
+                setCustomStrategyAddress(address)
+                // Ensure custom strategy is selected
+                if (selectedStrategyIndex === -1) {
+                  setSelectedStrategy(customStrategy(address))
+                }
+              }}
+            />
+          }
+          index={-1}
+          strategy={customStrategy(customStrategyAddress)}
+          selected={selectedStrategyIndex === -1}
+          onSelectBallot={() =>
+            setSelectedStrategy(customStrategy(customStrategyAddress))
+          }
+        />
+        <Button
+          htmlType="submit"
+          type="primary"
+          disabled={
+            selectedStrategy === undefined ||
+            (selectedStrategyIndex === -1 && // custom strategy
+              (!customStrategyAddress ||
+                !utils.isAddress(customStrategyAddress)))
+          }
+          onClick={() => {
```

Comment:
**suggestion**: I think it might make sense to wrap this whole drawer component in a `form` and leverage the form submit behavior rather than button `onClick`. WDYT?

Reply:
If we did this it'd mean getting rid of the states and using the form for everything, a solid overhaul. Not worth imo just to remove an `onClick`